pymeasure/experiment/results.py

```diff
@@ -92,6 +92,42 @@ def format_header(self):
         return self.delimiter.join(self.columns)
 
 
+class CSVFormatter_Pandas(logging.Formatter):
+    """ Formatter of data results """
+
+    def __init__(self, columns, delimiter=',', line_break='\n'):
+        """Creates a csv formatter for a given list of columns (=header).
+
+        :param columns: list of column names.
+        :type columns: list
+        :param delimiter: delimiter between columns.
+        :type delimiter: str
+        """
+        super().__init__()
+        self.columns = columns
+        self.delimiter = delimiter
+        self.line_break = line_break
+
+    def format(self, record):
+        """Formats a record as csv.
+
+        :param record: record to format.
+        :type record: pandas.DataFrame
+        :return: a string
+        """
+        return record.to_csv(
+            sep=self.delimiter,
+            header=False,
+            index=False,
+            # explicit line_terminator required, otherwise Windows
+            # uses \r\n which results in double blank lines
+            line_terminator=self.line_break
+        )
+
+    def format_header(self):
+        return self.delimiter.join(self.columns)
```

Comment:
I'm not sure either... but since in the future eventually all file handling is supposed to be done by pandas, let's have it take care of the header column as well. I agree.